tests_func/load/mongo_load.go

```diff
@@ -0,0 +1,247 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/tinsane/tracelog"
+	testUtils "github.com/wal-g/wal-g/tests_func/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"io/ioutil"
+	"sync"
+)
+
+var contextByName = map[string]context.Context{
+	"todo":       context.TODO(),
+	"background": context.Background(),
+}
+
+type MongoWorker struct {
+	wg sync.WaitGroup
+	ch chan interface{}
+	fs []func()
+}
+
+func CreateMongoWorker(chSz int) MongoWorker {
+	return MongoWorker{ch: make(chan interface{}, chSz)}
+}
+
+func (mw *MongoWorker) AddMongoOp(client *mongo.Client, sop string) error {
+	var arr []json.RawMessage
+	var err error
+	err = json.Unmarshal([]byte(sop), &arr)
+	if err != nil {
+		return err
+	}
+	for _, ropdata := range arr {
+		var opdata map[string]json.RawMessage
+		err = json.Unmarshal(ropdata, &opdata)
+		if err != nil {
+			return fmt.Errorf("command expected to be a map, but found %s", testUtils.GetStringType(opdata))
+		}
+		switch string(opdata["op"]) {
+		case `"c"`:
+			commandDoc, err := parseCommandOp(opdata)
+			if err != nil {
+				return err
+			}
+			mw.mongoRunCommandOp(client, commandDoc)
+		case `"t"`:
+			transactionDoc, err := parseTransactionOp(opdata)
+			if err != nil {
+				return err
+			}
+			mw.mongoRunTransaction(client, transactionDoc)
+		default:
+			return fmt.Errorf("unknown command %v", string(opdata["op"]))
+		}
+	}
+	return nil
+}
+
+type MongoRunCommandOp struct {
+	DbName   string `json:"db"`
+	Ctx      string `json:"ctx"`
+	MakeLogs bool   `json:"makelogs"`
+	Doc      bson.D
+}
+
+func parseCommandOp(cd map[string]json.RawMessage) (*MongoRunCommandOp, error) {
+	var res MongoRunCommandOp
+	bmrco, err := json.Marshal(cd)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse opts for commad: %+v", err)
+	}
+	err = json.Unmarshal(bmrco, &res)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse opts for commad: %+v", err)
+	}
+	err = bson.UnmarshalExtJSON(cd["dc"], true, &res.Doc)
+	if err != nil {
+		return nil, err
+	}
+	return &res, nil
+}
+
+func mongoRunCommandOp(client *mongo.Client, op *MongoRunCommandOp) {
+	db := client.Database(op.DbName)
+	var result bson.M
+	err := db.RunCommand(contextByName[op.Ctx], op.Doc).Decode(&result)
+	if op.MakeLogs {
```

Comment:
Do we need this flag? We can log anything using "DebugLogger" and set logging level during initialization

Reply:
Fixed, now all logs are gone in error channel or result channel